### PR TITLE
New notebooks - Maplotlib basic and intermediate

### PR DESCRIPTION
**Pull request by srcarter3**
-added a new notebook and preview image for 'Matplotlib: Basics'
-added another new notebook and preview image for 'Matplotlib: Intermediate'
-note: both notebooks will need to be updated when we have working notebook preview pages

### CODE REVIEW

**Comment by lesserwhirls** [comment]
Looks good to me! I'll have @dopplershift take a look given his deep matplotlib background.

File: notebooks/visualization/matplotlib-intermediate.ipynb [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a href=\"#top\">Top</a>\n",
+    "\n",
+    "---\n",
+    "\n",
+    "## 2. Multiple Plots on One Figure\n",
+    "\n",
+    "The [Matlibplot: Basics]() notebook explained how to create a figure and add a plot to that figure.  It is also possible to add multiple plots to the same figure.  In this step, a figure is first created and then multiple plots are added, each with their own titles, axes, etc."
```
Comment (dopplershift) [inline notebooks/visualization/matplotlib-intermediate.ipynb:103]: I'd add an explanation about how to add multiple plots, like:
"This is accomplished by calling `fig.add_subplot()`. This function lays out multiple plots in a grid, by passing in the number of rows, number of columns, and the number of the plot within the grid. So for example, if we wanted to create a plot in the lower left corner of a (hypothetical) grid of plots with 2 rows and 3 columns, we could write `fig.add_subplot(2, 3, 4)`, where 4 means the 4th plot, counting across rows and down columns."

File: notebooks/visualization/matplotlib-intermediate.ipynb [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a href=\"#top\">Top</a>\n",
+    "\n",
+    "---\n",
+    "\n",
+    "## 3. Colorizing Scatter Plots\n",
+    "\n",
+    "Previously, in the [Matlibplot: Basics]() notebook, an example for drawing scatter plots was provided by setting the **linestyle** to non, and adding 'o' markers.  Another alternative is to use the **scatter** methods, while these are slower, they allow more control of the display of the data.  In this case, the data points will be colorized individually based upon a third variable."
```
Comment (dopplershift) [inline notebooks/visualization/matplotlib-intermediate.ipynb:154]: "non" -> "none"

Comment (dopplershift) [reply notebooks/visualization/matplotlib-intermediate.ipynb:154]: Also, maybe "allow more control" -> "allow more visualization options, like varying the color or size of the markers".

**Review by dopplershift — COMMENTED** [review]
Overall these look good. Just a few suggestions.

File: notebooks/visualization/matplotlib-intermediate.ipynb (hunk 2 above)
Comment (srcarter3) [reply notebooks/visualization/matplotlib-intermediate.ipynb]: Added these changes with updated PR

File: notebooks/visualization/matplotlib-intermediate.ipynb (hunk 1 above)
Comment (srcarter3) [reply notebooks/visualization/matplotlib-intermediate.ipynb]: Added this change with updated PR

File: notebooks/visualization/matplotlib-intermediate.ipynb [hunk 3]
```diff
@@ -100,7 +100,8 @@
     "\n",
     "## 2. Multiple Plots on One Figure\n",
     "\n",
-    "The [Matlibplot: Basics]() notebook explained how to create a figure and add a plot to that figure.  It is also possible to add multiple plots to the same figure.  In this step, a figure is first created and then multiple plots are added, each with their own titles, axes, etc."
+    "The Matlibplot: Basics notebook explained how to create a figure and add a plot to that figure.  It is also possible to add multiple plots to the same figure.  In this step, a figure is first created and then multiple plots are added, each with their own titles, axes, etc.  Each plot is added using the *add_subplot()* function, which lays out multiple plots in a grid by passing the number of rows, columns, and which plot within the grid (starting with 1, not 0).\n",
```
Comment (dopplershift) [inline notebooks/visualization/matplotlib-intermediate.ipynb:103]: Matlibplot -> Matplotlib

File: notebooks/visualization/matplotlib-intermediate.ipynb [hunk 4]
```diff
@@ -151,7 +152,7 @@
     "\n",
     "## 3. Colorizing Scatter Plots\n",
     "\n",
-    "Previously, in the [Matlibplot: Basics]() notebook, an example for drawing scatter plots was provided by setting the **linestyle** to non, and adding 'o' markers.  Another alternative is to use the **scatter** methods, while these are slower, they allow more control of the display of the data.  In this case, the data points will be colorized individually based upon a third variable."
+    "Previously, in the Matlibplot: Basics notebook, an example for drawing scatter plots was provided by setting the **linestyle** to none, and adding 'o' markers.  Another alternative is to use the **scatter** methods, while these are slower, they allow more visualization options of the data, such as style or color of the individual markers.  In this case, the data points will be colorized individually based upon a third variable."
```
Comment (dopplershift) [inline notebooks/visualization/matplotlib-intermediate.ipynb:155]: Matlibplot -> Matplotlib

Comment (srcarter3) [reply notebooks/visualization/matplotlib-intermediate.ipynb]: fixed the misspelling, thank you for pointing that out!

File: notebooks/visualization/matplotlib-intermediate.ipynb (hunk 3 above)
Comment (srcarter3) [reply notebooks/visualization/matplotlib-intermediate.ipynb]: fixed the misspelling, thanks!

File: notebooks/visualization/matplotlib-basics.ipynb [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Imports\n",
+    "\n",
+    "The first step is to set up the notebook so that matlibplot plots appear inline as images.  Then we import the necessary packages for this notebook."
```
Comment (dopplershift) [inline notebooks/visualization/matplotlib-basics.ipynb:51]: matplotlib

File: notebooks/visualization/matplotlib-intermediate.ipynb [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+    "* [matplotlib.pyplot.figure](https://matplotlib.org/3.3.3/api/_as_gen/matplotlib.pyplot.figure.html)\n",
+    "* [matplotlib.pyplot.axes](https://matplotlib.org/3.3.3/api/_as_gen/matplotlib.pyplot.axes.html)\n",
+    "* [matplotlib.pyplot.imshow](https://matplotlib.org/3.3.3/api/_as_gen/matplotlib.pyplot.imshow.html)\n",
+    "* [matplotlib.pyplot.contour](https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.contour.html)\n",
+    "* [matplotlib.pyplot.contourf](https://matplotlib.org/3.3.3/api/_as_gen/matplotlib.pyplot.contourf.html)\n",
+    "\n",
+    "\n",
+    "### Related Notebooks\n",
+    "\n",
+    "* Matlibplot: Basics\n",
```
Comment (dopplershift) [inline notebooks/visualization/matplotlib-intermediate.ipynb:316]: Here too

**Review by dopplershift — CHANGES_REQUESTED** [review]
Sorry, I didn't do the search for "matlibplot" before, but just did now and found a couple more.

File: notebooks/visualization/matplotlib-intermediate.ipynb (hunk 6 above)
Comment (srcarter3) [reply notebooks/visualization/matplotlib-intermediate.ipynb]: Sorry, I should have done that sooner myself.  I went through and searched for 'matlibplot' in all of my notebooks and found a few more misspellings as well, that are fixed in the latest PR.

File: notebooks/visualization/matplotlib-basics.ipynb (hunk 5 above)
Comment (srcarter3) [reply notebooks/visualization/matplotlib-basics.ipynb]: Thank you!

File: notebooks/visualization/matplotlib-intermediate.ipynb (hunk 6 above)
Comment (dopplershift) [reply notebooks/visualization/matplotlib-intermediate.ipynb]: No worries!

**Review by dopplershift — APPROVED** [review]

**Pull request merged**